core/src/banking_stage.rs

```diff
@@ -706,54 +701,33 @@ impl BankingStage {
         let mut proc_start = Measure::start("consume_buffered_process");
-        let reached_end_of_slot;
         let num_packets_to_process = unprocessed_transaction_storage.len();
 
         // TODO: Right now we iterate through buffer and try the highest weighted transaction once
         // but we should retry the highest weighted transactions more often.
-        let (bank_start, poh_recorder_lock_time) = measure!(
-            poh_recorder.read().unwrap().bank_start(),
-            "poh_recorder.read",
-        );
-        slot_metrics_tracker.increment_consume_buffered_packets_poh_recorder_lock_us(
-            poh_recorder_lock_time.as_us(),
+        let returned_payload = unprocessed_transaction_storage.process_packets(
+            bank_start.working_bank.clone(),
+            banking_stage_stats,
+            slot_metrics_tracker,
+            |packets_to_process, payload| {
+                Self::do_process_packets(
+                    bank_start,
+                    payload,
+                    recorder,
+                    transaction_status_sender,
+                    gossip_vote_sender,
+                    banking_stage_stats,
+                    qos_service,
+                    log_messages_bytes_limit,
+                    &mut consumed_buffered_packets_count,
+                    &mut rebuffered_packet_count,
+                    &test_fn,
+                    packets_to_process,
+                )
+            },
         );
 
-        if let Some(BankStart {
-            working_bank,
-            bank_creation_time,
-        }) = bank_start
-        {
-            let returned_payload = unprocessed_transaction_storage.process_packets(
-                working_bank.clone(),
-                banking_stage_stats,
-                slot_metrics_tracker,
-                |packets_to_process, payload| {
-                    Self::do_process_packets(
-                        max_tx_ingestion_ns,
-                        &working_bank,
-                        &bank_creation_time,
-                        payload,
-                        recorder,
-                        transaction_status_sender,
-                        gossip_vote_sender,
-                        banking_stage_stats,
-                        qos_service,
-                        log_messages_bytes_limit,
-                        &mut consumed_buffered_packets_count,
-                        &mut rebuffered_packet_count,
-                        &test_fn,
-                        packets_to_process,
-                    )
-                },
-            );
-
-            reached_end_of_slot = returned_payload.reached_end_of_slot;
-        } else {
-            reached_end_of_slot = true;
-        }
-
-        if reached_end_of_slot {
+        if returned_payload.reached_end_of_slot {
             slot_metrics_tracker.set_end_of_slot_unprocessed_buffer_len(
                 unprocessed_transaction_storage.len() as u64,
```

Comment:
`unprocessed_transaction_storage.process_packets()` doesn't seem to check `end_of_slot` for early exist. Wondering if this metrics report still retain its original intention?

Reply:
Should probably also be setting this in the case we don't have a working bank due to expiration.

Given the above comment on the previous loop doing it chunk-by-chunk, this change probably makes the metric more accurate, right? It was previously possible different chunks are using different working banks (and thus slots) I think.